README.md

```diff
@@ -429,4 +429,115 @@ No token or bad token response
 }
 ```
 
+### Contacts
 
+Get lognin credentials: <br>
+`Refer to Companies "Get login credentials" above`
+
+#### Get all contacts for a user
+Request:
+
+```
+GET /api/v1/contacts
```

Comment:
In order for this path to work as intended and reflect the schema, would it make sense to structure the request as going through the individual user id to access contacts? The way it's currently set up leads me to believe that this index action will access all contacts regardless of who they belong to.

I envision the path looking like: GET /api/v1/users/user:id/contacts

Reply:
Great question Joe!
Contacts are set up similarly to companies. This design choice was made because users log in through sessions, which generate a token. That token contains the user’s ID, allowing Rails to determine which user is making the request. When the token is passed in, Rails automatically associates it with the correct user, so there’s no need to explicitly include the user ID in the path. This is why contacts and companies are accessed directly under their respective endpoints without nesting under the user ID-basically it eliminates the need for /users/:id in the routes.

Given this approach, I think job applications could also benefit from following the same structure as contacts and companies for consistency and to leverage the token-based user identification. Nesting this resource under the user ID in the routes seems redundant since the token already identifies the user making the request.

Let me know if you’d like to discuss this further. Perhaps we need to talk about this in our standup?

Reply:
I think Candice did a great job explaining this!
Additionally, with the user ID already present in the authorization token, it seems redundant to include it in the URL. This also gives the code a cleaners and more uniform structure, and allows for additional information potentially needed for querying to be stored within the token.
In my reading I have found that token-based authentication is a more widely adopted standard, whereas including user IDs in URLs is becoming less common due to its limitations and potential risks.